service/internal/security/standard_crypto.go

```diff
@@ -241,11 +251,7 @@ func (s StandardCrypto) RSAPublicKey(kid string) (string, error) {
 }
 
 func (s StandardCrypto) ECCertificate(kid string) (string, error) {
-	ecKeys, ok := s.keys[AlgorithmECP256R1]
-	if !ok || len(ecKeys) == 0 {
-		return "", ErrCertNotFound
-	}
-	k, ok := ecKeys[kid]
+	k, ok := s.keysByID[AlgorithmECP256R1]
```

Comment:
This looks like a bug, should this be lookup by id?

Reply:
yeah and a good argument for more unit test coverage